server/src/main/java/org/opensearch/index/codec/CodecService.java

```diff
@@ -73,10 +74,20 @@ public CodecService(@Nullable MapperService mapperService, IndexSettings indexSe
             codecs.put(BEST_COMPRESSION_CODEC, new Lucene99Codec(Mode.BEST_COMPRESSION));
             codecs.put(ZLIB, new Lucene99Codec(Mode.BEST_COMPRESSION));
         } else {
-            codecs.put(DEFAULT_CODEC, new PerFieldMappingPostingFormatCodec(Mode.BEST_SPEED, mapperService, logger));
-            codecs.put(LZ4, new PerFieldMappingPostingFormatCodec(Mode.BEST_SPEED, mapperService, logger));
-            codecs.put(BEST_COMPRESSION_CODEC, new PerFieldMappingPostingFormatCodec(Mode.BEST_COMPRESSION, mapperService, logger));
-            codecs.put(ZLIB, new PerFieldMappingPostingFormatCodec(Mode.BEST_COMPRESSION, mapperService, logger));
+            // CompositeCodec still delegates to PerFieldMappingPostingFormatCodec
+            // We can still support all the compression codecs when composite index is present
+            // hence we're defining the codecs like below
+            if (mapperService.isCompositeIndexPresent()) {
```

Comment:
I think we should put a check that only if feature flag is enabled, we will add the composite codecs.

Reply:
We block in mapping phase itself based on feature flag, is it needed here ?

Reply:
yeah, we should. I don't think these codecs should be loaded if the feature flag is disabled. If the feature flag is enabled, it would anyways need a restart of the process.
Also, these codecs are loaded at a shard level, its just additional objects loaded into memory.

Reply:
As discussed, since we have feature flag protection during index creation, we can skip adding it here.